docs/userguide/joins.rst

```diff
@@ -0,0 +1,260 @@
+Join Apps
+=========
+
+Join apps allows an app to define a sub-workflow: the app can launch other apps
+and incorporate them into the main task graph. They can be specified using the
+`join_app` decorator.
+
+Usage
+-----
+
+A `join_app` looks quite like a python app, but should return a ``Future``,
```

Comment:
do you mean `python_app` instead of python_app?  ie a parsl decorated python app or just a regular python app?

Reply:
whats a "regular python app"  that isn't parsl decorated?

Reply:
I meant that maybe this should be "`python_app`" instead of "python app"